File: app/src/main/java/com/example/recyclerviewhomework/adapter/GallaryViewHolder.java
package com.example.recyclerviewhomework.adapter;

import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.recyclerviewhomework.R;
import com.example.recyclerviewhomework.model.Gallery;

public class GallaryViewHolder extends RecyclerView.ViewHolder {

    public ImageView ivPicture;

    public GallaryViewHolder(@NonNull View itemView) {
        super(itemView);
        ivPicture = itemView.findViewById(R.id.ivPicture);
    }

    public void bindPicture(Gallery gallery){
        Glide.with(itemView.getContext())
                .load(gallery.getPicture()).dontAnimate().into(ivPicture);
    }
}

File: app/src/main/java/com/example/recyclerviewhomework/adapter/GallaryViewHolder.kt
package com.example.recyclerviewhomework.adapter

import androidx.recyclerview.widget.RecyclerView
import com.example.recyclerviewhomework.databinding.GalleryItemBinding
import com.example.recyclerviewhomework.model.Gallery
import com.example.recyclerviewhomework.presentation.IClickListener

class GallaryViewHolder(var binding: GalleryItemBinding, private val onLongItemClick: IClickListener<Gallery>) : RecyclerView.ViewHolder(binding.root) {

    fun bindPicture(gallery: Gallery) {
        itemView.setOnLongClickListener { onLongItemClick.onItemClick(gallery) }
        binding.gallery = gallery
        binding.executePendingBindings()
    }
}
